### ISSUE TITLE
Grafana 5.x

### ISSUE THREAD

**Issue by fadjar340** [pueteam/datasource-plugin-solr#1]
There are problem when I use this datasource using Grafana 5.0.x.
The datasource created produce error and the dashboard can not created.

There's some error in withCredential: undefined
also in the error message just said "type"undefined

Regards,
Fadjar

**Comment by ishtartec** [comment]
Hi @fadjar340,
can you post the configuration of the data source?. And the configuration of the component/graph you are trying to setup?.

Thanks!

**Comment by hienhn** [comment]
Hello, 
I have the same error with Grafana 5.1.0 and Solr 7.3.1.
Firstly, I can not use the command: "grafana-cli plugins install pue-solr-datasource" because
pue-solr-datasource is not found. Then, I download the .zip from github and put it directly in 
/usr/local/var/lib/grafana/plugins/pue-solr-datasource (I'm using a Mac).
After that, SolR appear in the list of data source of Grafana so that I can choose it and configure URL with server "http://localhost:8983/" without credentials. 
When creating a Grafana dashboard with Solr, I can see a list of solr collection but when I choose one of these collections, there is no data. I've checked with Solr to be sure that the data exists in the chosen collection. The dashboard error is "can not read property type of undefined".
Thanks for help.

**Comment by ishtartec** [comment]
Thanks for reporting @hienhn. I'll check and let you know.

**Comment by hienhn** [comment]
Ok, thanks for replying me. Juste to be more precise, as I'm new with Solr, I've followed the tutorial here to create 2 solr servers in 2 nodes (8983 and 7574) in cloud: 
https://lucene.apache.org/solr/guide/7_3/solr-tutorial.html
Then I create a collection and "post" a csv into it. Data appears in both nodes 8983 and 7574 but I can not visualize it with Grafana using the plugin.

**Comment by ishtartec** [comment]
@hienhn is there any timestamp field in your collection? Something like a tdate field. Remember that Grafana is a timeseries analytics dashboard, so the data sources like this one need a timestamp or date field.

**Comment by hienhn** [comment]
Yes, there is a field "processdata_timestamp" of format ["2018-05-22T00:04:31Z"]. With the same data, I can create dashboard with Elastic Search but not with Solr.
Thank you.

**Comment by ishtartec** [comment]
Hi @hienhn, can you update/redownload the repo? I've made some changes. There was a problem with field listing on Solr 7. Can you try again and let me know if it fixes your problem?.

**Comment by hienhn** [comment]
@ishtartec: Thank you very much for the update. I've installed the new plugin. It's much better now. I can choose my collections and my variables now in grafana. There is no more error "can not read property type of undefined" in the dashboard 👍 . However, I still can not visualize my data. It's not an error but it said "Data points outside time range".  In Grafana, I chose to see data from the last 6 months. My data is collected on 22 May 2018, the "processdata_timestamps" column is of format ["2018-05-22T00:04:31Z"]. Do you think that this format is understood by SolR? I think this is more a problem of SolR than the plugin Grafana now:)
